src/modules/job-manager/plugins/perilog.c

```diff
@@ -138,24 +138,41 @@ static void perilog_proc_delete (struct perilog_proc *proc)
     }
 }
 
-static void emit_finish_event (struct perilog_proc *proc, int status)
+static void emit_finish_event (struct perilog_proc *proc,
+                               flux_subprocess_t *sp)
 {
+    int status = flux_subprocess_status (sp);
     if (proc->prolog) {
         /*
          *  If prolog failed, raise job exception before prolog-finish
          *   event is emitted to ensure job isn't halfway started before
          *   the exception is raised:
          */
-        if (status != 0
-            && flux_jobtap_raise_exception (proc->p,
-                                            proc->id,
-                                            "prolog",
-                                            0,
-                                            "prolog failed with status=%d",
-                                            status) < 0)
-                flux_log_error (flux_jobtap_get_flux (proc->p),
-                                "prolog-finish: flux_jobtap_raise_exception");
-
+        if (status != 0) {
+            int code = flux_subprocess_exit_code (sp);
+            int sig;
+            char *errmsg;
+            if ((sig = flux_subprocess_signaled (sp)) > 0
+                || (sig = (code - 128)) > 0) {
+                asprintf (&errmsg,
+                          "prolog killed by signal %d%s",
+                          sig,
+                          sig == SIGTERM ? " (timeout or job canceled)" : "");
+            }
```

Comment:
I think this leaks `errmsg`?

Reply:
That is fixed in the latest push (sorry I pushed too early)

Reply:
Right then this is looking good!